app/src/main/java/com/hover/stax/transactions/TransactionHistoryAdapter.kt

```diff
@@ -8,16 +8,19 @@ import android.view.ViewGroup
 import android.view.LayoutInflater
 import android.view.View
 import androidx.core.text.HtmlCompat
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
 import com.hover.sdk.transactions.Transaction
+import com.hover.stax.channels.Channel
 import com.hover.stax.databinding.TransactionListItemBinding
 import java.util.Locale
 
-class TransactionHistoryAdapter(private var listOfTransactionActionPair: List<Pair<StaxTransaction, HoverAction?>>,
-                                private val selectListener: SelectListener) :
-	RecyclerView.Adapter<HistoryViewHolder>() {
+class TransactionHistoryAdapter(private var listOfTransactionHistory: List<TransactionHistory>,
+                                private val selectListener: SelectListener)
+	: ListAdapter<TransactionHistory, HistoryViewHolder>(diffUtil) {
 
-	fun updateData(listOfPair: List<Pair<StaxTransaction, HoverAction?>>) {
-		this.listOfTransactionActionPair = listOfPair
+	fun updateData(listOfHistory:  List<TransactionHistory>) {
```

Comment:
You don't need this method when working with `ListAdapter`. To load the data, call `adapter.submitList(yourData)` and it will take care of everything. In it's current state, the `DiffUtil` does nothing.

Reply:
Thanks for the correction